jobs/pacman-rule-engine-2.0/src/main/java/com/tmobile/pacman/executor/PolicyExecutor.java

```diff
@@ -55,49 +75,156 @@ public class PolicyExecutor {
+        policyExecutor.fetchPolicyWiseParams(args[0]);
+        ExecutorService executor = Executors.newFixedThreadPool(POLICY_THREAD_POOL_SIZE);
+        for (Map<String, String> policyParam : policyExecutor.policyWiseParamsList) {
+            String executionId = UUID.randomUUID().toString(); // this is the unique
+            String status = fetchLatestPolicyStatus(policyParam);
+            //String status = "ENABLED";
+            executor.execute(() -> {
+                try {
+                    if (policyExecutor.resources.isEmpty()
+                            || PacmanSdkConstants.POLICY_STATUS_DISABLED.equalsIgnoreCase(status)) {
+                        AnnotationPublisher annotationPublisher = new AnnotationPublisher();
+                        annotationPublisher.populateExistingIssuesForType(policyParam);
+                        annotationPublisher.closeDanglingIssues(
+                                policyParam.get(PacmanSdkConstants.DATA_SOURCE_KEY) + "_"
+                                        + policyParam.get(PacmanSdkConstants.TARGET_TYPE),
+                                policyParam.get(PacmanSdkConstants.TARGET_TYPE));
+                    } else {
+                        policyExecutor.run(policyParam, executionId);
+                    }
+                } catch (Exception e) {
+                    logger.error(DATA_ALERT_ERROR_STRING + JOB_NAME + " with job id "
+                            + policyParam.get(PacmanSdkConstants.POLICY_UUID_KEY) + ". ExecutionId is -> "
+                            + executionId);
+                }
+            });
+
+        }
+        executor.shutdown();
+        while (!executor.isTerminated()) {
+        }
+        policyDoneSNS(args[0]);
+        ProgramExitUtils.exitSucessfully();
+    }
+
+    private void fetchPolicyWiseParams(String requestJson) {
+        String policyDetailsUrl = CommonUtils.getEnvVariableValue(PacmanSdkConstants.POLICY_DETAILS_URL);
+        try {
+            String jsonResponse = CommonUtils.doHttpPost(policyDetailsUrl, requestJson);
+            policyWiseParamsList = CommonUtils.extractListOfMaps(jsonResponse);
+        } catch (Exception e) {
+            logger.error("failed in getting Policy list for  {}", requestJson);
+            logger.error(DATA_ALERT_ERROR_STRING + JOB_NAME +  "failed in getting Policy list for " + requestJson);
+        }
+    }
+
+    private void setResourcesList() {
+        long startTime = resetStartTime();
+        try {
+            resources = ESUtils.getResourcesFromEs(this.source, this.targetType, null, null);
+            logger.debug("got resources for evaluvation, total resources: " + resources.size());
+            logger.debug("timeTakenToFetchInventory {}", CommonUtils.getElapseTimeSince(startTime));
+        } catch (Exception e) {
+            logger.error("unable to get inventory for " + this.source + "--" + this.targetType, e);
+            ProgramExitUtils.exitWithError();
+        }
+
+    }

```

Comment:
Let's have consistency with such methods. Either use `fetch` or `set`. Not both.

Reply:
I prefer `set` personally in case you don't have personal preference :)

Reply:
Let's say this way - if method do return something then it should be `getBlablabla`, if it doesn't return but do change internal state of the class instance then it should be `setBlablabla`